bokeh/model.py

```diff
@@ -431,7 +431,15 @@ def js_link(self, attr, other, other_attr):
             raise ValueError("%r is not a property of other (%r)" % (other_attr, other))
 
         from bokeh.models import CustomJS
-        cb = CustomJS(args=dict(other=other), code="other.%s = this.%s" % (other_attr, attr))
+
+        if attr_selector is None:
+            cb = CustomJS(args=dict(other=other), code="other.%s = this.%s" % (other_attr, attr))
+        else:
+            if type(attr_selector) is str:
+                attr_selector = "\"%s\"" % attr_selector
+            else:
+                attr_selector = str(attr_selector)
+            cb = CustomJS(args=dict(other=other), code="other.%s = this.%s[%s]" % (other_attr, attr, attr_selector))
```

Comment:
Bokeh 2.0 is Python 3.6+ so I think all this (all cases) can be simplified with f-strings:
```python
selector = f"[{attr_selector!r}]" if attr_selector else ""
cb = CustomJS(args=dict(other=other), code=f"other.{other_attr} = this.{attr}{selector}"
```

Reply:
Oh awesome, also I need to add documentation for the argument! Every time I commit does the whole test suite need to run?

Reply:
The test suit will run on every push. I can cancel the current run now, since I am around. :)

Reply:
Ah ok, in that case did I follow the appropriate steps? First PR so very unsure haha. Will create a commit with the reviewed changes.

Reply:
You are doing great  :) This will be a nice feature for 2.0 so it is very appreciated. It had occurred to me before that `js_link` was not as useful as it should be but I never made an issue or thought about how to make it better until your post.

Reply:
Great! I hope I can get involved with other features too 😄